homeassistant/auth/providers/homeassistant.py

```diff
@@ -85,18 +80,19 @@ def validate_login(self, username: str, password: str) -> None:
                 found = user
 
         if found is None:
-            # Do one more compare to make timing the same as if user was found.
-            hmac.compare_digest(hashed, hashed)
+            # check a hash to make timing the same as if user was found
+            bcrypt.checkpw(b'foo',
```

Comment:
We'll need a migration path. We can only do the migration while a user is logging in. Verify with old pattern -> if ok, migrate to new salt.

Reply:
added